CMakeLists.txt

```diff
@@ -0,0 +1,180 @@
+
+    find_package(Python COMPONENTS Interpreter Development.Embed REQUIRED)
+
+    if (NOT "${Python_VERSION}" STREQUAL "${OpenAssetIO_Python_VERSION}")
+        message(
+            FATAL_ERROR
+            "Found Python '${Python_VERSION}' but OpenAssetIO was built against"
+            " '${OpenAssetIO_Python_VERSION}'"
+        )
+    endif ()
+
+    #-----------------------------------------------------------------------
+    # Python sources tests.
+
+    # Python sources test.
+    add_test(NAME python COMMAND ${Python_EXECUTABLE} ${PROJECT_SOURCE_DIR}/src/test.py)
+    set_tests_properties(python PROPERTIES ENVIRONMENT PYTHONPATH=${OpenAssetIO_Python_SITELIB})
+
+    #-----------------------------------------------------------------------
+    # Python bridge test (loading Python plugin system from C++).
+
+    add_executable(test.python.bridge src/test.python.bridge.cpp)
+    add_test(python.cpp.bridge test.python.bridge)
+    target_link_libraries(
+        test.python.bridge
+        PRIVATE
+        OpenAssetIO::openassetio-python-bridge
+        Python::Python
+    )
+
+    # Storage for list of environment variables to set before running
+    # the test.
+    set(_envvars)
+
+    # Add location of Python sources in the OpenAssetIO package to the
+    # PYTHONPATH.
+    list(APPEND _envvars "PYTHONPATH=${OpenAssetIO_Python_SITELIB}")
+
+    # The test has an embedded Python interpreter, but is missing all
+    # the required basic Python libraries, so we need to set PYTHONHOME
+    # to the source Python package to avoid nasty errors.
+    execute_process(
+        COMMAND ${Python_EXECUTABLE} -c "import sys; sys.stdout.write(sys.prefix)"
+        OUTPUT_VARIABLE Python_PREFIX
+    )
+    list(APPEND _envvars "PYTHONHOME=${Python_PREFIX}")
+
+    # Fix library search path on MacOS.
+    if (APPLE)
+        # The `Python::Python` target's link options are set to e.g
+        # `@rpath/Python.framework/Versions/3.9/Python` but all our
+        # target's (CMake generated) @rpaths already contain
+        # `Python.framework/Versions/3.9`. So add another RPATH with
+        # the framework subdirectories stripped. This then allows our
+        # test target to find the Python library.
+        set_target_properties(test.python.bridge PROPERTIES BUILD_RPATH ${Python_PREFIX}/../../..)
+    endif ()
+
+    # Augment the PATH on Windows to find Python library dlls.
+    if (WIN32)
+        if (DEFINED OpenAssetIO_BINARY_DIR)
+            list(APPEND _envvars.path ${OpenAssetIO_BINARY_DIR_NATIVE})
+        endif ()
+        foreach (path ${Python_RUNTIME_LIBRARY_DIRS})
+            file(TO_NATIVE_PATH "${path}" path_native)
+            list(APPEND _envvars.path ${path_native})
+        endforeach ()
+        # Frustratingly, $<SEMICOLON> doesn't work here, we have to find
+        # the magic number of backslashes instead.
+        list(JOIN _envvars.path \\\; _envvars.path)
+        list(APPEND _envvars "PATH=${_envvars.path}")
```

Comment:
Take a peek at the USD test_package, as have had to do something similar here, but fewer backslashes.
Had to distinguish between semi-colons for CMake list purposes, and semi-colons for Windows PATH separators.

Reply:
I believe the number of required backlashes corresponds to the number of times the variable is substituted.  Here there is the JOIN then the APPEND then the final ENVIRONMENT substitution.  I'm not sure at what point each backslash is collapsed, since it was more-or-less a trial-and-error thing.

Reply:
Sure, it was trial and error for me on USD too. Just wanted to point out that it had been tackled elsewhere if you want to take a look.